src/components/Home/OptionsForm.tsx

```diff
@@ -1,12 +1,60 @@
+import { TextInputSize } from '@commonComponents/TextInput/TextInput.d';
 import { OptionsFormContainer } from './Home.styles';
+import ColorInput from '@commonComponents/ColorInput/ColorInput';
+import TextInput from '@commonComponents/TextInput/TextInput';
+import FileInput from '@commonComponents/FileInput/FileInput';
+import { useState } from 'react';
+
+interface QRForm {
+  size: string;
+  color: string;
+  background: string;
+  file: File | undefined;
+}
 
 const OptionsForm = () => {
+  const [form, setForm] = useState<QRForm>({
+    size: '0px',
+    color: '',
+    background: '',
+    file: undefined,
+  });
+
+  // Handles changes in the form properties based on the input type
+  // TECH DEBT: Enhance this mechanism to handle different input types more gracefully
+  const handleChangeForm =
+    (propertyName: keyof QRForm) =>
+    (
+      event: React.ChangeEvent<
+        | HTMLInputElement
+        | HTMLTextAreaElement
+        | HTMLSelectElement
+        | HTMLInputElement
+      >,
+    ) => {
+      // Extract the input target and handle different input types accordingly
+      const target = event.target as HTMLInputElement;
+      setForm((prevForm) => ({
```

Comment:
I think this is overengineering. It would be more readable if we had one handler for every type.